flink-connectors/flink-jdbc/src/main/java/org/apache/flink/api/java/io/jdbc/JDBCLookupFunction.java

```diff
@@ -139,6 +139,14 @@ public void eval(Object... keys) {
 
 		for (int retry = 1; retry <= maxRetryTimes; retry++) {
 			try {
+				try {
```

Comment:
Check whether connection is valid on each writes is expensive. How about catching the "statement closed" exception (mentioned in FLINK-16681) and solving it by creating a new connection.